### PR TITLE
replace hardcoded rke2 to product var instead

### PR DESCRIPTION
**Pull request by aganesh-suse**
#### Proposed Changes ####

Upgrade Node replacement test fails today while looking up FetchToken since it has rke2 hardcoded. 
Updating FetchToken to have product arg and updated the call to the method as well. 

No

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

2. Verify code lint; we should not have errors.

3. Update the documentation if needed.

4. Update makefile and docker run if adding new tests.

5. Run your tests at least 4 times with all configurations needed and possible.

6. If needed test with different os types.

### CODE REVIEW

**Review by fmoral2 — DISMISSED** [review]

File: pkg/testcase/upgradenodereplacement.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func TestUpgradeReplaceNode(cluster *shared.Cluster, version string) {
 	shared.LogLevel("info", "Scp files to new server nodes done\n")
 
 	serverLeaderIP := cluster.ServerIPs[0]
```
Comment (mdrahman-suse) [inline pkg/testcase/upgradenodereplacement.go:48]: qq… we would also need to assign the new external ips to the cluster.ServerIPs, isnt't it? else it would show the previous ips...

File: pkg/testcase/upgradenodereplacement.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func TestUpgradeReplaceNode(cluster *shared.Cluster, version string) {
 	Expect(scpErr).NotTo(HaveOccurred(), scpErr)
 	shared.LogLevel("info", "Scp files to new server nodes done\n")
 
-	serverLeaderIP := cluster.ServerIPs[0]
+	serverLeaderIP := newExternalServerIps[0]
```
Comment (mdrahman-suse) [inline pkg/testcase/upgradenodereplacement.go:48]: @aganesh-suse sorry please revert this change, will revisit this

Comment (fmoral2) [reply pkg/testcase/upgradenodereplacement.go:48]: yes please

Comment (aganesh-suse) [reply pkg/testcase/upgradenodereplacement.go]: i suspect that it is talking to the old IP to get the token to copy on to the new set of nodes maybe. so i dont think we should be using the new IP for this line. reverting the change.

Comment (fmoral2) [reply pkg/testcase/upgradenodereplacement.go]: https://github.com/rancher/distros-test-framework/pull/149#discussion_r1723988444

**Review by fmoral2 — APPROVED** [review]

**Review by mdrahman-suse — APPROVED** [review]
LGTM
